compiler/src/test/kotlin/utility.kt

```diff
@@ -0,0 +1,104 @@
+package dk.aau.cs.d409f19
+
+import dk.aau.cs.d409f19.cellumata.CompilerData
+import dk.aau.cs.d409f19.cellumata.ast.*
+import dk.aau.cs.d409f19.cellumata.visitors.LiteralExtractorVisitor
+import dk.aau.cs.d409f19.cellumata.visitors.ScopeCheckVisitor
+import dk.aau.cs.d409f19.cellumata.visitors.TypeChecker
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+
+class Utilities {
+
+    companion object {
+
+        /**
+         * Compiles a program given as string parameter
+         */
+        fun compileProgram(program: String): CompilerData {
+            val source = CharStreams.fromString(program)
+            val lexer = dk.aau.cs.d409f19.antlr.CellmataLexer(source)
+            val tokenStream = CommonTokenStream(lexer)
+            val parser = dk.aau.cs.d409f19.antlr.CellmataParser(tokenStream)
+
+            // Build AST
+            val startContext = parser.start()
+            val ast = reduce(startContext)
+
+            // Extract literals
+            LiteralExtractorVisitor().visit(ast)
+
+            // Symbol table and scope
+            val scopeChecker = ScopeCheckVisitor()
+            scopeChecker.visit(ast)
+            val symbolTable = scopeChecker.getSymbolTable()
+
+            // Type checking
+            TypeChecker(symbolTable).visit(ast)
+
+            return CompilerData(parser, ast, scopeChecker)
```

Comment:
Why doesn't `CompilerData` include the symbol table? I can see else where that it is extracted the scope checker

Reply:
Fixed by having `CompilerData` contain the `Table` instead. Also makes sure that any changes during typechecking is included.

Reply:
Relevant commit: 414adfcb7570dd1b179cbd74b2abf1cc8929ebee